streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -605,7 +716,12 @@ public boolean test(final K1 key, final V1 value) {
         Objects.requireNonNull(selector, "selector can't be null");
         Objects.requireNonNull(serialized, "serialized can't be null");
         final SerializedInternal<KR, V> serializedInternal = new SerializedInternal<>(serialized);
-        String selectName = internalSelectKey(selector);
+        final String selectName = internalSelectKey(selector);
+
+        final StatelessProcessorNode<KR, V> graphNode = new StatelessProcessorNode<>(this.name + "GROUP_BY",
```

Comment:
Is it correct to set parent node name as `this.name + "GROUP_BY"`? Seems the parent node name is not set in this way. Ditto below.

Reply:
ack, fixed.